fdb-record-layer-core/src/test/java/com/apple/foundationdb/record/provider/foundationdb/indexes/MultidimensionalIndexTest.java

```diff
@@ -236,13 +239,13 @@ RecordMetaDataBuilder addAdditionalValueMultidimensionalIndex(@Nonnull final Rec
         return metaDataBuilder;
     }
 
-    public void loadRecords(@Nonnull final RecordMetaDataHook hook, final long seed, final List<String> calendarNames, final int numSamples) throws Exception {
+    public void loadRecords(@Nonnull final RecordMetaDataHook hook, final long seed, final List<String> calendarNames, final int numSamples)  {
```

Comment:
We might want to add a test that explicitly attempts to read and write to the index more-or-less concurrently. If I'm right that the `inner` cursor of an `AsyncLockCursor` can begin executing before the cursor is fully formed, then I think that that test would fail at least some of the time.

Reply:
You mean a test where a update happens asynchronously with other updates and a readCursor?

Reply:
Right, a test where one (or more) index updates happen at the same time as a read